src/benchmark.cc

```diff
@@ -684,505 +364,577 @@ void Benchmark::AddRange(std::vector<int>* dst, int lo, int hi, int mult) {
-    } else if (FLAGS_benchmark_repetitions > 1) {
-      name_field_width =
-          std::max<size_t>(name_field_width, benchmark.name.size() + 7);
-    } else {
-      name_field_width =
-          std::max<size_t>(name_field_width, benchmark.name.size());
-    }
+  int name_field_width = 10;
+  for (std::size_t i = 0; i < benchmarks.size(); i++) {
+    name_field_width = std::max<int>(name_field_width,
+                                    // Maybe add space for appending "_stddev"
+                                    FLAGS_benchmark_repetitions > 1
+                                    ? benchmarks[i].name.size() + 7
+                                    : benchmarks[i].name.size());
   }
 
   // Print header here
   BenchmarkReporter::Context context;
-  context.num_cpus = NumCPUs();
-  context.mhz_per_cpu = CyclesPerSecond() / 1000000.0f;
-  //  context.cpu_info = base::CompactCPUIDInfoString();
+  context.num_cpus = benchmark::NumCPUs();
+  context.mhz_per_cpu = benchmark::CyclesPerSecond() / 1000000.0f;
+
   context.cpu_scaling_enabled = CpuScalingEnabled();
   context.name_field_width = name_field_width;
 
-  if (reporter->ReportContext(context))
-    for (internal::Benchmark::Instance& benchmark : benchmarks)
-      Benchmark::RunInstance(benchmark, reporter);
+  if (reporter->ReportContext(context)) {
+    for (const auto& bench_instance : benchmarks) {
+      RunBenchmark(bench_instance, reporter);
+    }
+  }
 }
 
 void FindMatchingBenchmarkNames(const std::string& spec,
                                 std::vector<std::string>* benchmark_names) {
   if (spec.empty()) return;
 
-  std::vector<internal::Benchmark::Instance> benchmarks;
-  BenchmarkFamilies::GetInstance()->FindBenchmarks(spec, &benchmarks);
-  std::transform(benchmarks.begin(), benchmarks.end(), benchmark_names->begin(),
-                 [](const internal::Benchmark::Instance& b) { return b.name; });
+  std::vector<benchmark::internal::Benchmark::Instance> benchmarks;
+  benchmark::internal::Benchmark::FindBenchmarks(spec, &benchmarks);
+  for (const auto& bench_instance : benchmarks) {
+    benchmark_names->push_back(bench_instance.name);
+  }
 }
 
-}  // end namespace internal
+} // end namespace internal
+
 
-void RunSpecifiedBenchmarks(const BenchmarkReporter* reporter /*= nullptr*/) {
+void RunSpecifiedBenchmarks(BenchmarkReporter* reporter) {
   std::string spec = FLAGS_benchmark_filter;
-  if (spec.empty() || spec == "all")
-    spec = ".";  // Regexp that matches all benchmarks
-  internal::ConsoleReporter default_reporter;
-  internal::RunMatchingBenchmarks(
-      spec, reporter == nullptr ? &default_reporter : reporter);
+  if (spec.empty()) {
+    // Nothing to do
+  } else {
+    if (spec == "all") {
+      spec = ".";         // Regexp that matches all benchmarks
+    }
+    internal::ConsoleReporter default_reporter;
+    internal::RunMatchingBenchmarks(spec, reporter ? reporter : &default_reporter);
+    std::exit(0);
```

Comment:
why an explicit exit here? just let the function exit so people calling this aren't surprised.